main/src/ca/uwaterloo/flix/language/phase/jvm/GenChannelClasses.scala

```diff
@@ -0,0 +1,705 @@
+package ca.uwaterloo.flix.language.phase.jvm
+
+import ca.uwaterloo.flix.api.Flix
+import ca.uwaterloo.flix.language.ast.ExecutableAst.Root
+import org.objectweb.asm.Label
+import org.objectweb.asm.ClassWriter
+import org.objectweb.asm.Opcodes._
+
+/**
+  * Generates bytecode for the channel classes.
+  */
+object GenChannelClasses {
+  /**
+    * Returns the bytecode for the channel classes built-in to the Flix language.
+    */
+  def gen()(implicit root: Root, flix: Flix): Map[JvmName, JvmClass] = {
+
+    // Type that we need a channel class for
+    val types = List(JvmType.PrimBool, JvmType.PrimChar, JvmType.PrimFloat, JvmType.PrimDouble,
+      JvmType.PrimByte, JvmType.PrimShort, JvmType.PrimInt, JvmType.PrimLong, JvmType.Object)
+
+    // Generating each channel class
+    types.map{ tpe =>
+      val classType = JvmName.getChannelClassType(tpe)
+      classType.name -> JvmClass(classType.name, genChannelClass(classType, tpe))
+    }.toMap
+  }
+
+  /**
+    * Generating class `classType` with value of type `channelType`
```

Comment:
Please add some Java pseudo code that shows the overall structure of the generated class or shows an example for a specific type.